src/main/java/pl/edu/agh/toik/ec/starter/SimpleStarter.java

```diff
@@ -40,6 +45,7 @@ public void run() {
 
         Topology topology = topologyService.getTopology(configuration.getNamingService().getAgentsIds());
         configuration = configuration.withTopology(topology);
+        configuration = configuration.withEmigrateStrategy(emigrateStrategy);
```

Comment:
emigrateStrategy should be injected directly into Configuration, as it is possible to create it in the xml configuration. Only topology is handled in an other way, because the Topology service has to create it first, so topology is not known during bean injection. We will then avoid 1. passing another argument to the starter, 2. creating another configuration instance here. 3. adding another constructor to Configuration
But you don't have to worry about this, we can handle this as the starter and configuration component